### PR TITLE
codespell: workflow, config + some (quite a few) typos fixed

### PR DESCRIPTION
**Pull request by yarikoptic**
Probably the most  boring PR to review ;)

Individual commits might be easier to digest

### CODE REVIEW

**Comment by baskaryan** [comment]
this is awesome @yarikoptic! looks like the new codespell CI check is failing :') could we add a make command or script to make it easy to run codespell locally?

**Comment by baskaryan** [comment]
> right? ;) the point is that running `codespell` is sufficient if you have it installed "properly" (given that config is in pyproject.toml you need `tomli` installed too).

haha fair, but i think probably still helpful to have simple make command for other contributors. added to Makefile and contributor guidelines, if that looks ok to you happy to land!

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -123,6 +123,31 @@ This can be very helpful when you've made changes to only certain parts of the p
 
 We recognize linting can be annoying - if you do not want to do it, please contact a project maintainer, and they can help you with it. We do not want this to be a blocker for good code getting contributed.
 
+### Spellcheck
+
+Spellchecking for this project is done via [codespell](https://github.com/codespell-project/codespell).
```
Comment (yarikoptic) [inline .github/CONTRIBUTING.md:128]: note that `codespell` is not really a full blown spellchecker - it is rather a "finder of common typos", so many misspelled words could exist, and on contrary -- `codespell` could alert about some correct, but rarely used words, which are more typically appearing as typos.  So may be make it explicit here:

```suggestion
Spellchecking for this project is done via [codespell](https://github.com/codespell-project/codespell).
Note that `codespell` finds common typos, so could have false-positive (correctly spelled but rarely used) and false-negatives (not finding misspelled) words.
```

File: tests/unit_tests/tools/openapi/test_specs/zapier/apispec.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
...
... that is very popular on Zapier\n* reduces final result into about ~500 words\n\nTrimmed results are ideal for inserting directly back into the prompt context of a large language models without blowing up context token window limits.\n\nExample of a trimmed results payload from \"Gmail: Find Email\":\n\n```\n    {\n        \"result\": {\n            \"from__email\": \"mike@zapier.com\",\n            \"from__name\": \"Mike Knoop\",\n            \"subject\": \"Re: Getting setup\",\n            \"body_plain\": \"Hi Karla, thanks for following up. I can confirm I got access to everything! ... Thanks! Mike\",\n            \"cc__emails\": \"bryan@zapier.com, wade@zapier.com\"\n            \"to__email\": \"Mike Knoop\",\n        }\n    }\n```\n## Changelog <a name=\"changelog\"></a>\n\n**Mar 20, 2023**\nShipped two minor but breaking changes, and one other minor change to the API's response data:\n\n* Route: `/api/v1/configuration-link/`\n  * Key `url` is now `configuration_link` **(breaking change)**\n* Route: `/api/v1/exposed/{exposed_app_action_id}/execute/`\n  * Key `rating_url` is now `review_url` **(breaking change)**\n* Route: `/api/v1/exposed/`\n  * Added `configuration_link` key"
```
Comment (yarikoptic) [inline tests/unit_tests/tools/openapi/test_specs/zapier/apispec.json:6]: oh -- should may be this file be skipped?

Comment (baskaryan) [reply tests/unit_tests/tools/openapi/test_specs/zapier/apispec.json]: good point! will skip test data files

**Comment by yarikoptic** [comment]
Please let me know if you want me to do anything (e.g. to rebase/reapply codespell etc).

**Comment by baskaryan** [comment]
> Please let me know if you want me to do anything (e.g. to rebase/reapply codespell etc).

all lgtm, thanks @yarikoptic!!

**Pull request merged**